### PR TITLE
Avoid some allocations by using a fixed size vector

### PR DESCRIPTION
**Pull request by ZehMatt**
Also makes the code a bit easier to read. There are more places where we can use this, just one thing to note that its not just a pure flat array, it still keeps track of the count so this can not be replaced with raw array or std::array when the structure fields must remain POD.

### CODE REVIEW

File: src/openrct2/ride/Vehicle.h
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.h:159]: Do we need to use the using here could we forward declare the class then can avoid adding the header

File: src/openrct2/ride/Vehicle.cpp
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:919]: Will we do a follow on PR that moves all the sound updating into a separate class?

Comment (ZehMatt) [reply src/openrct2/ride/Vehicle.cpp:919]: Sounds like a good idea

Comment (IntelOrca) [reply src/openrct2/ride/Vehicle.cpp:919]: Perhaps wait until the music PR is merged first as that already moves the ride music out. Unless there should be RideAudio and VehicleAudio.

Comment (ZehMatt) [reply src/openrct2/ride/Vehicle.cpp:919]: I see, then I'll revert the vehicle part.

**Comment by Gymnasiast** [comment]
I guess this is also a performance thing? Could you include benchmarks?

**Comment by ZehMatt** [comment]
Initially it replaced the vector for the ride audio parameters but @IntelOrca is working on #13738 so I dropped the commit, there is no performance improvement right now. The vector is not just for performance but to also make the code a lot easier to grasp, there are a lot of areas where some sort of "list" is managed in fixed space, this is where the FixedVector can come in handy.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**
